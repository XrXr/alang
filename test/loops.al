main :: proc() {
    last := 0
    fib := 1
    i := 0
    for i < 55 {
        next := fib + last
        last = fib
        fib = next
        i = i + 1
    }

    fib2last := 0
    fib2 := 1
    for 1..55 {
        next := fib2 + fib2last
        fib2last = fib2
        fib2 = next
    }

    puts("should be 0\n")
    print_int(fib-fib2)
}
