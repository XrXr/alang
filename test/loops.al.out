should be 0
0
